spec/reek/smells/irresponsible_module_spec.rb

```diff
@@ -61,6 +66,80 @@ class BadClass
     expect(src).to reek_of :IrresponsibleModule, name: 'Foo::Bar'
   end
 
+  it 'does not report modules used only as namespaces' do
+    src = <<-EOS
+      module Foo
+        # Describes Bar
+        class Bar
+          def baz
+          end
+        end
+      end
+    EOS
+    expect(src).not_to reek_of(:IrresponsibleModule)
+  end
+
+  it 'does not report classes used only as namespaces' do
+    src = <<-EOS
+      class Foo
+        # Describes Bar
+        module Bar
+          def qux
+          end
+        end
+      end
+    EOS
+    expect(src).not_to reek_of(:IrresponsibleModule)
+  end
+
+  it 'reports modules that have both nested modules and methods' do
+    src = <<-EOS
+      module Foo
+        def foofoo
+        end
+        # Describes Bar
+        module Bar
+        end
+      end
+    EOS
+    expect(src).to reek_of(:IrresponsibleModule)
+  end
+
+  it 'reports modules that have both nested modules and singleton methods' do
+    src = <<-EOS
+      module Foo
+        def self.foofoo
+        end
+        # Describes Bar
+        module Bar
+        end
+      end
+    EOS
+    expect(src).to reek_of(:IrresponsibleModule)
+  end
+
+  it 'reports modules that have both nested modules and methods on the singleton class' do
+    src = <<-EOS
+      module Foo
+        class << self
+          def foofoo
+          end
+        end
+        # Describes Bar
+        module Bar
+        end
+      end
+    EOS
+    expect(src).to reek_of(:IrresponsibleModule)
+  end
+
+  it 'foo' do
```

Comment:
Hm, I’m not sure that it shuld go – I think it would be beneficial to have explicit spec saying whether comment-less
```Ruby
class CompilationError < RuntimeError; end
```
and
```Ruby
CompilationError = Class.new(RuntimeError)
```
smell of IrresponsibleModule or not.